driver.go

```diff
@@ -12,49 +12,80 @@ import (
 	"github.com/alexbrainman/odbc/api"
 )
 
+const (
+	DriverPoolModeNone  DriverPoolMode = 0
+	DriverPoolModeBasic DriverPoolMode = 1
+	DriverPoolModeFull  DriverPoolMode = 2
+)
+
+type DriverPoolMode int
+
 var drv Driver
 
 type Driver struct {
 	Stats
-	h       api.SQLHENV // environment handle
-	initErr error
+	h        api.SQLHENV // environment handle
+	initErr  error
+	poolMode DriverPoolMode
+}
+
+func (d *Driver) PoolMode() DriverPoolMode {
+	return d.poolMode
+}
+
+func (d *Driver) IsPooling() bool {
+	return d.poolMode != DriverPoolModeNone
+}
+
+func (d *Driver) IsFullPooling() bool {
+	return d.poolMode == DriverPoolModeFull
+}
+
+func (d *Driver) Close() error {
+	// TODO(brainman): who will call (*Driver).Close (to dispose all opened handles)?
+	h := d.h
+	d.h = api.SQLHENV(api.SQL_NULL_HENV)
+	return releaseHandle(h)
 }
 
 func initDriver() error {
 
+	//TODO: find a way to make this attribute changeable at runtime
+	//Enable connection pooling (this should be executed before allocating the environment handle)
+	ret := api.SQLSetEnvUIntPtrAttr(api.SQLHENV(api.SQL_NULL_HENV), api.SQL_ATTR_CONNECTION_POOLING, api.SQL_CP_ONE_PER_HENV, api.SQL_IS_UINTEGER)
+	if IsError(ret) {
+		drv.poolMode = DriverPoolModeNone
+	} else {
+		drv.poolMode = DriverPoolModeBasic
+	}
+
 	//Allocate environment handle
 	var out api.SQLHANDLE
 	in := api.SQLHANDLE(api.SQL_NULL_HANDLE)
-	ret := api.SQLAllocHandle(api.SQL_HANDLE_ENV, in, &out)
+	ret = api.SQLAllocHandle(api.SQL_HANDLE_ENV, in, &out)
 	if IsError(ret) {
 		return NewError("SQLAllocHandle", api.SQLHENV(in))
 	}
 	drv.h = api.SQLHENV(out)
 	err := drv.Stats.updateHandleCount(api.SQL_HANDLE_ENV, 1)
 	if err != nil {
+		drv.Close()
 		return err
 	}
 
 	// will use ODBC v3
 	ret = api.SQLSetEnvUIntPtrAttr(drv.h, api.SQL_ATTR_ODBC_VERSION, api.SQL_OV_ODBC3, 0)
 	if IsError(ret) {
-		defer releaseHandle(drv.h)
+		defer drv.Close()
 		return NewError("SQLSetEnvUIntPtrAttr(SQL_ATTR_ODBC_VERSION, SQL_OV_ODBC3)", drv.h)
 	}
 
-	//TODO: find a way to make this attribute changeable at runtime
-	//Enable connection pooling
-	ret = api.SQLSetEnvUIntPtrAttr(drv.h, api.SQL_ATTR_CONNECTION_POOLING, api.SQL_CP_ONE_PER_HENV, api.SQL_IS_UINTEGER)
-	if IsError(ret) {
-		defer releaseHandle(drv.h)
-		return NewError("SQLSetEnvUIntPtrAttr(SQL_ATTR_CONNECTION_POOLING, SQL_CP_ONE_PER_HENV)", drv.h)
-	}
-
-	//Set relaxed connection pool matching
-	ret = api.SQLSetEnvUIntPtrAttr(drv.h, api.SQL_ATTR_CP_MATCH, api.SQL_CP_RELAXED_MATCH, api.SQL_IS_UINTEGER)
-	if IsError(ret) {
-		defer releaseHandle(drv.h)
-		return NewError("SQLSetEnvUIntPtrAttr(SQL_ATTR_CP_MATCH, SQL_CP_RELAXED_MATCH)", drv.h)
+	if drv.IsPooling() {
+		//Set relaxed connection pool matching
+		ret = api.SQLSetEnvUIntPtrAttr(drv.h, api.SQL_ATTR_CP_MATCH, api.SQL_CP_RELAXED_MATCH, api.SQL_IS_UINTEGER)
+		if !IsError(ret) {
```

Comment:
Remove next 3 lines, since drv.poolMode won't be used.

Reply:
Please read my comment above, regarding the pool mode constants.